File: 3.Platform_Solutions/AlgoExpert/Arrays/02-Validate-Subsequence/Python/solution.py
def isValidSubsequence(array, sequence):
    # Write your code here.
    currSequenceIdx = 0
    
    for number in array:
        if sequence[currSequenceIdx] == number:
            currSequenceIdx += 1
    
        if currSequenceIdx == len(sequence):
            return True
    
    return False

File: 3.Platform_Solutions/AlgoExpert/Arrays/02-Validate-Subsequence/Golang/solution.go
package main

func IsValidSubsequence(array []int, sequence []int) bool {
	// Write your code here.
	currSequenceIdx := 0

	for _, number := range array {
		if sequence[currSequenceIdx] == number {
			currSequenceIdx++
		}

		if currSequenceIdx == len(sequence) {
			return true
		}
	}
	return false
}
